### PR TITLE
hyper-boring: update to hyper 1.0.0

### PR DESCRIPTION
**Pull request by howardjohn**
This PR upgrades hyper-boring to support hyper 1.0.0-rc.3. Most of this is by moving dependencies from hyper core to hyper-util, as much of the functionality we depend on has been removed from the core.

### CODE REVIEW

**Comment by Noah-Kennedy** [comment]
Thanks for doing this!

We'll have to wait until hyper 1.0 is fully released to merge and release this, but as soon as that happens I'm happy to cut a release.

File: Cargo.toml [hunk 1]
```diff
@@ -36,8 +36,11 @@ futures = "0.3"
 tokio = "1"
 anyhow = "1"
 antidote = "1.0.0"
-http = "0.2"
-hyper = { version = "0.14", default-features = false }
+http = "1.0"
+hyper = { version = "1.0", default-features = false }
+hyper-util = { git = "https://github.com/hyperium/hyper-util" }
```
Comment (shahn) [inline Cargo.toml:41]: This adds a dependency on a not-released crate. Personally, I feel like the update to hyper 1.0 should wait for a release of hyper-util

Comment (howardjohn) [reply Cargo.toml:41]: hyper-util is now published, moved to official release

Comment (shahn) [reply Cargo.toml]: awesome!

**Comment by howardjohn** [comment]
Unfortunately I messed this PR up and needed to reopen as https://github.com/cloudflare/boring/pull/194

**Comment by howardjohn** [comment]
superseded by https://github.com/cloudflare/boring/pull/194

**Pull request closed without merging**